### PR TITLE
feat(ndt): remove ndt package

### PR DESCRIPTION
**Pull request by kminoda**
## Description
Remove `ndt` package.

The package was intended to provide a unique interface for various types of NDT implementation (like ndt_omp or pcl ones), but in real current Autoware is designed to use `ndt_omp` implementation. (e.g. the parameters are only designed for ndt_omp). In addition, as far as we know, NDT implementations other than `ndt_omp` does not seem to be used.

Thus, I propose to remove `ndt` and `ndt_pcl_modified` packages to simplify the localization module and especially `ndt_scan_matcher`.

## Related links
[[proposal] Removing NDT implementations other than ndt_omp](https://github.com/orgs/autowarefoundation/discussions/2936)

## Tests performed
I have ran Autoware with the sample data from tutorial, and confirmed that the localization module works fine.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by KYabuuchi** [comment]
Thank you for great PR!!
I built and tested your branch, and the NDT worked successfully.
I'll check out the details soon.

**Comment by KYabuuchi** [comment]
* There is still a description about OMP in [ndt_scan_matcher/config/ndt_scan_matcher.param.yaml#L25](https://github.com/kminoda/autoware.universe/blob/refactor/ndt/remove_package/localization/ndt_scan_matcher/config/ndt_scan_matcher.param.yaml#L25).
image


* There is no default value for `converged_param_type` in `tier4_localization_launch/config/ndt_scan_matcher.param.yaml`.
(On the other hand, `ndt_scan_matcher/config/ndt_scan_matcher.param.yaml` has the default value)


* There is no description of `converged_type` in the `ndt_scan_matcher/README.md`.
If you don't mind, could you add something like following? 
> Type of termination criteria for iterative calculations (0=TRANSFORM_PROBABILITY, 1=NEAREST_VOXEL_TRANSFORMATION_LIKELIHOOD)"

File: localization/ndt_scan_matcher/include/ndt_scan_matcher/ndt_scan_matcher_core.hpp [hunk 1]
```diff
@@ -76,32 +76,23 @@ struct NdtResult
...
 
 class NDTScanMatcher : public rclcpp::Node
 {
   using PointSource = pcl::PointXYZ;
   using PointTarget = pcl::PointXYZ;
+  using NormalDistributionsTransform =
+    pclomp::NormalDistributionsTransform<PointSource, PointTarget>;
 
   // TODO(Tier IV): move file
   struct OMPParams
```
Comment (KYabuuchi) [inline localization/ndt_scan_matcher/include/ndt_scan_matcher/ndt_scan_matcher_core.hpp:98]: I think we no longer need a OMPParams.

**Review by KYabuuchi — CHANGES_REQUESTED** [review]

File: localization/ndt_scan_matcher/include/ndt_scan_matcher/ndt_scan_matcher_core.hpp (hunk 1 above)
Comment (kminoda) [reply localization/ndt_scan_matcher/include/ndt_scan_matcher/ndt_scan_matcher_core.hpp]: Thanks for all the comments!

For `converged_param` comments: https://github.com/autowarefoundation/autoware.universe/pull/2053/commits/9ae2705cb7f47f84ab250401dcbfe5734529806c
For OMPParams comment: https://github.com/autowarefoundation/autoware.universe/pull/2053/commits/0248009f64b6efcf2c1fe96b3a533759f8f06eaa

**Review by KYabuuchi — APPROVED** [review]
LGTM

**Comment by KYabuuchi** [comment]
@YamatoAndo I am not the owner on this branch and cannot give final approval. 
Could you please review this PR in your free time? 

 (In the `main` branch I am the owner of `ndt_scan_matcher` , but this PR is derived from before that.)

**Comment by kminoda** [comment]
I think it depends on this one
https://github.com/autowarefoundation/autoware.universe/pull/2118

**Comment by kminoda** [comment]
Oh, I think this PR is blocked since @.YamatoAndo is the only code owner of `ndt` package (which will be removed with this PR).

@YamatoAndo Would you also approve this PR?

**Pull request merged**